src/test/compile-fail/lifetime-inference-give-expl-lifetime-param.rs

```diff
@@ -39,8 +39,7 @@ struct Cat<'x, T> { cat: &'x isize, t: T }
 struct Dog<'y> { dog: &'y isize }
 
 fn cat2<'x, 'y>(x: Cat<'x, Dog<'y>>) -> &'x isize {
-//~^ HELP: consider using an explicit lifetime parameter as shown: fn cat2<'x>(x: Cat<'x, Dog<'x>>) -> &'x isize
-    x.t.dog //~ ERROR: cannot infer
+    x.t.dog //~ ERROR E0312
```

Comment:
If you didn't skip on including the message, it would be clearer what's going on, btw.